### PR TITLE
MIG-1561: Release Notes for MTC 1.7.16

### PR DESCRIPTION
**Pull request by anarnold97**
### Jira

* [MIG-1561](https://issues.redhat.com/browse/MIG-1561)

### Version(s):

* OCP 4.12 → branch/enterprise-4.12
* OCP 4.13 → branch/enterprise-4.13
* OCP 4.14 → branch/enterprise-4.14
* OCP 4.15 → branch/enterprise-4.15
* OCP 4.16 → branch/enterprise-4.16

### Link to docs preview:

* [Migration Toolkit for Containers 1.7.16 release notes](https://76487--ocpdocs-pr.netlify.app/openshift-enterprise/latest/migration_toolkit_for_containers/release_notes/mtc-release-notes-1-7.html#migration-mtc-release-notes-1-7-16_mtc-release-notes)

### QE review:
- [ ] QE has approved this change.

Additional information:

### CODE REVIEW

File: modules/migration-mtc-release-notes-1-7-16.adoc [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+:_mod-docs-content-type: REFERENCE
+[id="migration-mtc-release-notes-1-7-16_{context}"]
+= {mtc-full} 1.7.16 release notes
+
+[id="resolved-issues-1-7-16_{context}"]
+== Resolved issues
+
+This release has the following resolved issues:
+
+.CVE-2023-45290: Golang: `net/http`: Memory exhaustion in `Request.ParseMultipartForm`
```
Comment (anarnold97) [inline modules/migration-mtc-release-notes-1-7-16.adoc:13]:
```suggestion
.CVE-2023-45290: Golang: `net/http`: Memory exhaustion in the `Request.ParseMultipartForm` method
```

File: modules/migration-mtc-release-notes-1-7-16.adoc [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+= {mtc-full} 1.7.16 release notes
+
+[id="resolved-issues-1-7-16_{context}"]
+== Resolved issues
+
+This release has the following resolved issues:
+
+.CVE-2023-45290: Golang: `net/http`: Memory exhaustion in the `Request.ParseMultipartForm` method
+
+A flaw was found in the `net/http` Golang standard library package, which impacts earlier versions of {mtc-short}. When parsing a `multipart` form, either explicitly with `Request.ParseMultipartForm` or implicitly with `Request.FormValue`, `Request.PostFormValue`, or `Request.FormFile`, limits on the total size of the parsed form are not applied to the memory consumed while reading a single form line. This permits a maliciously crafted input containing long lines to cause the allocation of arbitrarily large amounts of memory, potentially leading to memory exhaustion.
```
Comment (anarnold97) [inline modules/migration-mtc-release-notes-1-7-16.adoc:15]:
```suggestion
A flaw was found in the `net/http` Golang standard library package, which impacts earlier versions of {mtc-short}. When parsing a `multipart` form, either explicitly with `Request.ParseMultipartForm` or implicitly with `Request.FormValue`, `Request.PostFormValue`, or `Request.FormFile` methods, limits on the total size of the parsed form are not applied to the memory consumed while reading a single form line. This permits a maliciously crafted input containing long lines to cause the allocation of arbitrarily large amounts of memory, potentially leading to memory exhaustion.
```

**Review by rayfordj — APPROVED** [review]
/lgtm

File: modules/migration-mtc-release-notes-1-7-16.adoc [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+
+
+[id="known-issues-1-7-16_{context}"]
+== Known issues
+
+This release has the following known issues:
+
+.Direct Volume Migration is failing as the Rsync pod on the source cluster goes into an `Error` state
+
+On migrating any application with Persistent Volume Claim (PVC), the `Stage` migration operation succeeds with warnings, and Direct Volume Migration (DVM) fails with the `rsync` pod on the source namespace going into an `error` state. link:https://bugzilla.redhat.com/show_bug.cgi?id=2256141[(BZ#2256141)]
```
Comment (anarnold97) [inline modules/migration-mtc-release-notes-1-7-16.adoc:69]:
```suggestion
On migrating any application with a Persistent Volume Claim (PVC), the `Stage` migration operation succeeds with warnings, and Direct Volume Migration (DVM) fails with the `rsync` pod on the source namespace going into an `error` state. link:https://bugzilla.redhat.com/show_bug.cgi?id=2256141[(BZ#2256141)]
```

File: modules/migration-mtc-release-notes-1-7-16.adoc [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+
+This release has the following known issues:
+
+.Direct Volume Migration is failing as the Rsync pod on the source cluster goes into an `Error` state
+
+On migrating any application with a Persistent Volume Claim (PVC), the `Stage` migration operation succeeds with warnings, and Direct Volume Migration (DVM) fails with the `rsync` pod on the source namespace going into an `error` state. link:https://bugzilla.redhat.com/show_bug.cgi?id=2256141[(BZ#2256141)]
+
+.The conflict condition is briefly cleared after it is created
+
+When creating a new state migration plan that results in a conflict error message, the error message is cleared shortly after it is displayed. link:https://bugzilla.redhat.com/show_bug.cgi?id=2144299[(BZ#2144299)]
```
Comment (anarnold97) [inline modules/migration-mtc-release-notes-1-7-16.adoc:73]:
```suggestion
When creating a new state migration plan that returns a conflict error message, the error message is cleared very shortly after it is displayed. link:https://bugzilla.redhat.com/show_bug.cgi?id=2144299[BZ#2144299]
```

File: modules/migration-mtc-release-notes-1-7-16.adoc [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+
+
+[id="known-issues-1-7-16_{context}"]
+== Known issues
+
+This release has the following known issues:
+
+.Direct Volume Migration is failing as the Rsync pod on the source cluster goes into an `Error` state
+
+On migrating any application with a Persistent Volume Claim (PVC), the `Stage` migration operation succeeds with warnings, and Direct Volume Migration (DVM) fails with the `rsync` pod on the source namespace going into an `error` state. link:https://bugzilla.redhat.com/show_bug.cgi?id=2256141[(BZ#2256141)]
```
Comment (anarnold97) [inline modules/migration-mtc-release-notes-1-7-16.adoc:69]:
```suggestion
On migrating any application with a Persistent Volume Claim (PVC), the `Stage` migration operation succeeds with warnings, and the Direct Volume Migration (DVM) fails with the `rsync` pod on the source namespace moving into an `error` state. link:https://bugzilla.redhat.com/show_bug.cgi?id=2256141[(BZ#2256141)]
```

**Review by stillalearner — APPROVED** [review]
lgtm

File: modules/migration-mtc-release-notes-1-7-16.adoc [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+
+A flaw was found in the `webpack-dev-middleware package`, which impacts earlier versions of {mtc-short}. This flaw fails to validate the supplied URL address sufficiently before returning local files, which could allow an attacker to craft URLs to return arbitrary local files from the developer's machine.
+
+To resolve this issue, upgrade to {mtc-short} 1.7.16.
+
+For more details, see link:https://access.redhat.com/security/cve/cve-2024-29180[CVE-2024-29180].
+
+.CVE-2024-30255: `envoy`: HTTP/2 CPU exhaustion due to CONTINUATION frame flood
+
+A flaw was found in how the `envoy` proxy implements the HTTP/2 codec, which impacts earlier versions of {mtc-short}. There are insufficient limitations placed on the number of `CONTINUATION` frames that can be sent within a single stream, even after exceeding the header map limits of `envoy`. This flaw could allow an unauthenticated remote attacker to send packets to vulnerable servers. These packets could use up compute resources to cause a denial of service (DoS).
```
Comment (bhardesty) [inline modules/migration-mtc-release-notes-1-7-16.adoc:55]:
```suggestion
A flaw was found in how the `envoy` proxy implements the HTTP/2 codec, which impacts earlier versions of {mtc-short}. There are insufficient limitations placed on the number of `CONTINUATION` frames that can be sent within a single stream, even after exceeding the header map limits of `envoy`. This flaw could allow an unauthenticated remote attacker to send packets to vulnerable servers. These packets could consume compute resources and cause a denial of service (DoS).
```

File: modules/migration-mtc-release-notes-1-7-16.adoc [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+
+
+[id="known-issues-1-7-16_{context}"]
+== Known issues
+
+This release has the following known issues:
+
+.Direct Volume Migration is failing as the Rsync pod on the source cluster goes into an `Error` state
+
+On migrating any application with a Persistent Volume Claim (PVC), the `Stage` migration operation succeeds with warnings, and the Direct Volume Migration (DVM) fails with the `rsync` pod on the source namespace moving into an `error` state. link:https://bugzilla.redhat.com/show_bug.cgi?id=2256141[BZ#2256141]
```
Comment (bhardesty) [inline modules/migration-mtc-release-notes-1-7-16.adoc:69]:
```suggestion
On migrating any application with a Persistent Volume Claim (PVC), the `Stage` migration operation succeeds with warnings, but the Direct Volume Migration (DVM) fails with the `rsync` pod on the source namespace moving into an `error` state. link:https://bugzilla.redhat.com/show_bug.cgi?id=2256141[(BZ#2256141)]
```
In the previous "Known issues" sections, the BZ links are bolded. For consistency, you might consider doing that here as well.

File: modules/migration-mtc-release-notes-1-7-16.adoc [hunk 8]
```diff
@@ -0,0 +1,77 @@
...
+
+.The conflict condition is briefly cleared after it is created
+
+When creating a new state migration plan that returns a conflict error message, the error message is cleared very shortly after it is displayed. link:https://bugzilla.redhat.com/show_bug.cgi?id=2144299[BZ#2144299]
```
Comment (bhardesty) [inline modules/migration-mtc-release-notes-1-7-16.adoc:73]:
```suggestion
When creating a new state migration plan that returns a conflict error message, the error message is cleared very shortly after it is displayed. link:https://bugzilla.redhat.com/show_bug.cgi?id=2144299[(BZ#2144299)]
```

File: modules/migration-mtc-release-notes-1-7-16.adoc [hunk 9]
```diff
@@ -0,0 +1,77 @@
...
+
+.Migration fails when there are multiple Volume Snapshot Locations of different provider types configured in a cluster
+
+When there are multiple Volume Snapshot Locations (VSLs) in a cluster with different provider types, but you have not set any of them as the default VSL, Velero results in a validation error that causes migration operations to fail. link:https://bugzilla.redhat.com/show_bug.cgi?id=2180565[(BZ#2180565]
```
Comment (bhardesty) [inline modules/migration-mtc-release-notes-1-7-16.adoc:77]:
```suggestion
When there are multiple Volume Snapshot Locations (VSLs) in a cluster with different provider types, but you have not set any of them as the default VSL, Velero results in a validation error that causes migration operations to fail. link:https://bugzilla.redhat.com/show_bug.cgi?id=2180565[(BZ#2180565)]
```

**Review by bhardesty — COMMENTED** [review]
Looks good! I just had a couple very minor textual suggestions.

**Comment by ShaunaDiaz** [comment]
@anarnold97 This is lgtm to merge, with the exception that anything going into the `main` branch must be picked to the latest version as well, so the `4.17` branch. If that is not OK and the change should only go in the 4.12-4.16 branches, please use the `4.16` base branch and we can CP backwards.

**Comment by ShaunaDiaz** [comment]
Picking to 4.17 per Slack OK.

**Pull request merged**